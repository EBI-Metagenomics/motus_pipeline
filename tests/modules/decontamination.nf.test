nextflow_process {

    name "Test module Decontamination"
    script "modules/decontamination.nf"
    process "DECONTAMINATION"

    test("Decontamination PE") {

        when {
            params {
                reads = "${baseDir}/tests/modules/fixtures/bwa/"
                mode = "paired"
                name = "mgtest"
            }
            process {
                """
                input[0] = channel.fromFilePairs("${params.reads}/${params.name}_{1,2}.fastq.gz", checkIfExists: true).map { it[1] }
                input[1] = channel.fromPath(params.reference_genome)
                input[2] = channel.value(params.reference_genome_name)
                input[3] = channel.value(params.mode)
                input[4] = channel.value(params.name)
                """
            }
        }

        then {
            assert process.success
            assert path(process.out.decontaminated_reads.get(0).get(0)).linesGzip.size() == 1972
            assert path(process.out.decontaminated_reads.get(0).get(0)).linesGzip.contains("CCTGATTGTATGCAGTGGAGTCTATGCAGCGAGGCCAAATCATCCCGATTGAATGCTGTGGATTATATGGAGCATGGCCCAAATCATCCCGATTGAATGCAGTGGATTCCATGCAGCGTGATTGATTGCAGTGTAATCCATGCAGCGTTG")  // content of read_1
            assert path(process.out.decontaminated_reads.get(0).get(1)).linesGzip.size() == 1972
            assert path(process.out.decontaminated_reads.get(0).get(1)).linesGzip.contains("CAAACACGCTGCATGGAATCCACTGCATTCAATCGGGATGCTTTGGGCCATGCTGGATAATATCCACTGCATTCAATCAGGATGCATTGGGCCCCGCTGCCTGGAATCCACTGCATTCAACCGCAGTGCATGGAAGCCACTGCATTCAAT") // content of read_2
        }

    }
    test("Decontamination PE reverse input order") {

        when {
            params {
                reads = "${baseDir}/tests/modules/fixtures/bwa/"
                mode = "paired"
                name = "mgtest"
            }
            process {
                """
                input[0] = channel.fromFilePairs("${params.reads}/${params.name}_{2,1}.fastq.gz", checkIfExists: true).map { it[1] }
                input[1] = channel.fromPath(params.reference_genome)
                input[2] = channel.value(params.reference_genome_name)
                input[3] = channel.value(params.mode)
                input[4] = channel.value(params.name)
                """
            }
        }

        then {
            assert process.success
            assert path(process.out.decontaminated_reads.get(0).get(0)).linesGzip.size() == 1972
            assert path(process.out.decontaminated_reads.get(0).get(0)).linesGzip.contains("CCTGATTGTATGCAGTGGAGTCTATGCAGCGAGGCCAAATCATCCCGATTGAATGCTGTGGATTATATGGAGCATGGCCCAAATCATCCCGATTGAATGCAGTGGATTCCATGCAGCGTGATTGATTGCAGTGTAATCCATGCAGCGTTG")  // content of read_1
            assert path(process.out.decontaminated_reads.get(0).get(1)).linesGzip.size() == 1972
            assert path(process.out.decontaminated_reads.get(0).get(1)).linesGzip.contains("CAAACACGCTGCATGGAATCCACTGCATTCAATCGGGATGCTTTGGGCCATGCTGGATAATATCCACTGCATTCAATCAGGATGCATTGGGCCCCGCTGCCTGGAATCCACTGCATTCAACCGCAGTGCATGGAAGCCACTGCATTCAAT") // content of read_2
        }

    }
    test("Decontamination SE") {

        when {
            params {
                name = "mgtest_merged"
                reads = "${baseDir}/tests/modules/fixtures/bwa/"
                mode = "single"
            }
            process {
                """
                input[0] = channel.fromPath("${params.reads}/${params.name}.fastq.gz", checkIfExists: true)
                input[1] = channel.fromPath(params.reference_genome)
                input[2] = channel.value(params.reference_genome_name)
                input[3] = params.mode
                input[4] = params.name
                """
            }
        }

        then {
            assert process.success
            assert path(process.out.decontaminated_reads.get(0)).linesGzip.size() == 0
        }

    }
}
