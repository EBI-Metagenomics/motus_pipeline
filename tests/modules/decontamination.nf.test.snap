{
    "Decontamination SE": {
        "content": [
            {
                "decontaminated_reads": [
                    "mgtest_merged_clean.fastq.gz:md5,0d1779a98411057a7df60549a8ba94a0"
                ]
            }
        ],
        "timestamp": "2023-02-28T11:31:20+0000"
    },
    "Decontamination PE": {
        "content": [
            {
                "decontaminated_reads": [
                    [
                        "mgtest_clean_1.fastq.gz:md5,11f4cb5684a3ac52167f9392e99d81b6",
                        "mgtest_clean_2.fastq.gz:md5,69c29131e4f6d81f719a30109aa7cbc4"
                    ]
                ]
            }
        ],
        "timestamp": "2023-02-28T11:31:20+0000"
    },
    "Decontamination PE reverse input order": {
        "content": [
            {
                "decontaminated_reads": [
                    [
                        "mgtest_clean_1.fastq.gz:md5,a979047936858df8934584646cd819d9",
                        "mgtest_clean_2.fastq.gz:md5,1746997336f7f145181b547549934bbe"
                    ]
                ]
            }
        ],
        "timestamp": "2023-02-28T11:31:20+0000"
    }
}